initrd/bin/mount-usb

```diff
@@ -34,56 +108,102 @@ if [ -z "$(cat /tmp/usb_block_devices)" ]; then
 fi
 
 USB_MOUNT_DEVICE=""
-# Check for the common case: a single USB disk with one partition
-if [ `cat /tmp/usb_block_devices | wc -l` -eq 1 ]; then
-  USB_MOUNT_DEVICE=`cat /tmp/usb_block_devices`
-fi
-# otherwise, let the user pick
-if [ -z ${USB_MOUNT_DEVICE} ]; then
-  > /tmp/usb_disk_list
-  for i in `cat /tmp/usb_block_devices`; do
-    echo $i $(blkid | grep $i | grep -o 'LABEL=".*"' | cut -f2 -d '"') >> /tmp/usb_disk_list
-  done
+# Check if the user has specified a USB device
+if [ -n "$DEVICE" ]; then
+  DEBUG "Checking if $DEVICE is a USB device"
+  if grep -q $DEVICE /tmp/usb_block_devices; then
```

Comment:
Not sure why this one got flagged as "outdated" since this code has not changed - I still suggest reworking list_usb_devices and testing just the device you care about rather than the roundabout "enumerate everything and test, then check if my device was in the results" with the potential problems relating to the naïve grep